pkg/resource/clickpipe.go

```diff
@@ -1275,13 +1291,43 @@ func (c *ClickPipeResource) Create(ctx context.Context, request resource.CreateR
 		}
 	}
 
-	expectedState := api.ClickPipeRunningState
+	// Determine expected state(s) based on configuration
+	var stateCheckFunc func(string) bool
 	if plan.Stopped.ValueBool() {
-		expectedState = api.ClickPipeStoppedState
+		stateCheckFunc = func(state string) bool {
+			return state == api.ClickPipeStoppedState
+		}
+	} else {
+		// Check if this is a snapshot-only Postgres pipe
+		isSnapshotOnly := false
+		var sourceModel models.ClickPipeSourceModel
+		if diags := plan.Source.As(ctx, &sourceModel, basetypes.ObjectAsOptions{}); diags == nil {
+			if !sourceModel.Postgres.IsNull() {
+				var postgresSource models.ClickPipePostgresSourceModel
+				if diags := sourceModel.Postgres.As(ctx, &postgresSource, basetypes.ObjectAsOptions{}); diags == nil {
+					if !postgresSource.Settings.IsNull() {
+						var settings models.ClickPipePostgresSettingsModel
+						if diags := postgresSource.Settings.As(ctx, &settings, basetypes.ObjectAsOptions{}); diags == nil {
+							isSnapshotOnly = settings.ReplicationMode.ValueString() == api.ClickPipePostgresReplicationModeSnapshot
+						}
+					}
+				}
+			}
+		}
+
+		// For snapshot-only pipes, accept both Running and Completed states
+		if isSnapshotOnly {
+			stateCheckFunc = func(state string) bool {
+				return state == api.ClickPipeRunningState || state == api.ClickPipeCompletedState
+			}
+		} else {
+			stateCheckFunc = func(state string) bool {
+				return state == api.ClickPipeRunningState
+			}
+		}
 	}
-	if _, err := c.client.WaitForClickPipeState(ctx, serviceID, createdClickPipe.ID, func(state string) bool {
-		return state == expectedState
-	}, clickPipeStateChangeMaxWaitSeconds); err != nil {
+
+	if _, err := c.client.WaitForClickPipeState(ctx, serviceID, createdClickPipe.ID, stateCheckFunc, clickPipeStateChangeMaxWaitSeconds); err != nil {
```

Comment:
What would be the golden path for CDC pipes with a 1h-1d snapshot?

Reply:
Moving to follow up PR.